inventory_management_system_api/schemas/catalogue_category.py

```diff
@@ -91,10 +91,9 @@ def validate_unit(cls, unit_value: Optional[str], info: ValidationInfo) -> Optio
             raise ValueError(f"Unit not allowed for boolean catalogue item property '{info.data['name']}'")
         return unit_value
 
-    @field_validator("allowed_values")
     @classmethod
-    def validate_allowed_values(
-        cls, allowed_values: Optional[AllowedValuesSchema], info: ValidationInfo
+    def check_valid_allowed_values(
+        cls, allowed_values: Optional[AllowedValuesSchema], catalogue_category_data: dict[str, Any]
```

Comment:
This is the data of the property so `catalogue_category_data` should be updated.